admittance_controller/src/admittance_controller.cpp

```diff
@@ -27,6 +27,23 @@
 
 namespace admittance_controller
 {
+
+static geometry_msgs::msg::Wrench add_wrenches(
+  const geometry_msgs::msg::Wrench & a, const geometry_msgs::msg::Wrench & b)
+{
+  geometry_msgs::msg::Wrench res;
+
+  res.force.x = a.force.x + b.force.x;
+  res.force.y = a.force.y + b.force.y;
+  res.force.z = a.force.z + b.force.z;
+
+  res.torque.x = a.torque.x + b.torque.x;
+  res.torque.y = a.torque.y + b.torque.y;
+  res.torque.z = a.torque.z + b.torque.z;
+
+  return res;
+}
+
```

Comment:
Do we need additional subscriber to add these wrench offsets?. Does #1215 work for your case? Where the offsets are added at configure time and then it exposes state interfaces, these interfaces can be used by the admittance controller

Reply:
The admittance controller reads the FT values directly via the corresponding state interfaces and not via a ros topic. I do not see how a change in the FTBroadcaster might help here.

Additionally adding an additional subscriber would allow to publish certain wrench profiles. I would guess the parameter system is way to slow for that.

Reply:
Got it. May be I misunderstood your description sorry. You mean to have this offsets in the wrench, so the admittance control can apply a certain force in that specific direction?

Reply:
Exactly. Sorry if my description was misleading